heliopy/data/util.py

```diff
@@ -238,6 +237,15 @@ def cdf_units(cdf_, manual_units=None):
                 warnings.warn(message, Warning)
         except KeyError:
             continue
+        try:
+            if len(cdf_[key][0]) > 1:
+                val = []
+                val.append(key)
+                for x in range(0, len(cdf_[key][0])):
+                    field = key + "{}".format('_' + str(x))
+                    val.append(field)
+        except Exception as e:
```

Comment:
Why do you need to catch an exception here?

Reply:
Some columns don't have length attribute, so it passes an error, like `nHe2` gives an
`len() of unsized object`

Reply:
Could you use something like this code that I put in recently instead?

```python
            key_shape = cdf[cdf_key].shape
            ndims = len(key_shape)
            if ndims == 1:
                df[df_key] = cdf[cdf_key][...]
            elif ndims == 2:
                for i in range(key_shape[1]):
                    df[df_key + '_' + str(i)] = cdf[cdf_key][...][:, i]
```
The first if block handles single columns, the second if block handles multiple columns in one variable. I think in general it's better to not catch exceptions if possible.

Reply:
That sounds way better! Thanks!